lessons/r/ggplot2/ggplot2.Rmd

```diff
@@ -0,0 +1,423 @@
+
+```{r, first-ggplot-with-plus}
+# assign plot to a variable
+diamonds_plot <- ggplot(diamonds, aes(x = carat, y = price))
+
+# draw the plot
+diamonds_plot + geom_point()
+```
+
+Notes:
+
+- Anything you put in the `ggplot()` function can be seen by any geom layers
+that you add (i.e., these are universal plot settings). This includes the x and
+y axis you set up in `aes()`.
+- You can also specify aesthetics for a given geom independently of the
+aesthetics defined globally in the `ggplot()` function.
+- The `+` sign used to add layers must be placed at the end of each line containing
+a layer. If, instead, the `+` sign is added in the line before the other layer,
+`ggplot2` will not add the new layer and R will return an error message.
+
+
+### Building plots iteratively + scatter plots
+
+Building plots with `ggplot2` is typically an iterative process. We start by
+defining the dataset we'll use, lay the axes, and choose a geom:
+  
+```{r}
+ggplot(diamonds, aes(x = carat, y = price)) +
+  geom_point()
+```
+
+Then, we start modifying this plot to extract more information from it. For
+instance, we can add transparency (`alpha`) to reduce overplotting:
+  
+  
+```{r}
+ggplot(diamonds, aes(x = carat, y = price)) +
+  geom_point(alpha = 0.5)
+```
+
+From the eye test, we can see some evidence of clusters in this scatter plots.
+It's possible that one of the categorical variables we have in the data could
+explain this pattern. Coloring the data points according to a categorical
+variable is an easy way to assess if there seems to be correlation. Let's try
+this with `cut`.
+
+```{r}
+ggplot(diamonds, aes(x = carat, y = price, color = cut)) +
+  geom_point(alpha = 0.5)
+```
+
+While some clustering is visible here, the pattern is not especially clear.
+Perhaps something else in our dataset might be a better predictor. Let's try
+`clarity` instead:
+
+```{r}
+ggplot(diamonds, aes(x = carat, y = price, color = clarity)) +
+geom_point(alpha = 0.5)
+```
+
+Turns out clarity appears to be a strong predictor here after all.  Given that
+IF1 is the best possible clarity measurement, we see that even 1-2 carat IF1
+diamonds fetch equivalent prices to 4-5 carat I1 (lowest clarity) diamonds.
+
+### Combining `dplyr` and `ggplot` - line plots
+
+What if we wanted to see how average carat varies by price?
+We'd first have to calculate what the mean carat value is for each price
+before plotting. Fortunately for us, we can bring back some of those `dplyr`
+skills from our previous lesson to do just that.
+
+```{r}
+diamonds %>% 
+  group_by(price) %>% 
+  summarise(mean_carat = mean(carat))
+```
+
+This transformed dataset can be piped right into a `ggplot` call, with which
+we can proceed to plot:
+
+```{r}
+diamonds %>% 
+  group_by(price) %>% 
+  summarise(mean_carat = mean(carat)) %>% 
+  ggplot(aes(x = price, y = mean_carat)) +
+  geom_line()
+```
+
+There's definitely a lot going on here -- let's just focus on the price range 0 - 5000:
```

Comment:
Maybe mention that you don't need to pass the dataframe to the ggplot call when it being piped from dplyr.